embedding.py

```diff
@@ -0,0 +1,47 @@
+import argparse
+import os
+import time
+
+import numpy as np
+import pandas as pd
+from sentence_transformers import SentenceTransformer
+
+
+def encode(sentences: np.ndarray, model: str = "all-MiniLM-L6-v2") -> None:
+    """Generates the embeddings of sentences using a
+    sentence-transformers model."""
+
+    model = SentenceTransformer("sentence-transformers/" + model)
+    embeddings = model.encode(sentences)
+
+    return embeddings
+
+
+def get_df_from_json(filepath: str) -> pd.DataFrame:
+    return pd.read_json(filepath, lines=True)
+
+
+def main():
+    # filepath = "static/arxiv-small.json"
+    parser = argparse.ArgumentParser()
```

Comment:
argparser only makes sense when running this as a script, so you should move this down to the `if __name__=="__main__"` section and take `filename` as an argument.

I think that `main` should receive the loaded dataframe, so you have call `get_df_from_json` before calling it and pass in the result.

Finally, `main` is not a very descriptive name. Function names are important too!

Reply:
Done, check changes